### DISCUSSION TITLE
Missing Artifacts Tracking Issue

### DISCUSSION THREAD

**Discussion by liesauer** [nulastudio/NetCoreBeauty#36] [General]
**PLEASE REPORT MISSING ARTIFACTS IN HERE.**

**在反馈前，你可以尝试启用`BeautyGitCDN`，对于中国大陆来说，这应该一直保持使用`Gitee`镜像。**

https://github.com/nulastudio/NetBeauty2/blob/8c0ac96c30fc29c841929f0cd7bf379cbf356923/README.md?plain=1#L73-L74

<br /><br />

Normally, you can see the build log saying something like this:
```
Artifact does not exist. v5.0.17/win-x64
```

Please report in the following format:
```
FXRVersion: vX.Y.Z
RID: OS-BIT
```

**Comment by Greendq** [comment]
FXRVersion v6.0.7/win-x64 
RID win-x64

The same for linux-x64 and macos-x64 after latest Visual Studio 2022 update to version 17.2.6.

**Reply by liesauer** [reply]
already uploaded.

**Comment by roo20** [comment]
Artifact does not exist. v6.0.8/win-x64

after .NET 6,0,8 Security Update KB5016990

**Reply by liesauer** [reply]
uploaded

**Comment by Greendq** [comment]
FXRVersion v6.0.9/win-x64
RID win-x64

The same for linux-x64 and macos-x64 after latest Visual Studio 2022 update to version 17.3.4.

**Reply by liesauer** [reply]
uploaded

**Reply by Greendq** [reply]
Great, thank you!

**Comment by Greendq** [comment]
FXRVersion v6.0.11/win-x64
RID win-x64

The same for linux-x64 and macos-x64 after latest Visual Studio 2022 update to version 17.4.0

**Reply by liesauer** [reply]
uploaded

**Comment by Greendq** [comment]
Thank you!

**Comment by z0y1** [comment]
Artifact does not exist. v7.0.2/win-x64

**Reply by liesauer** [reply]
this should exists.
![image](https://user-images.githubusercontent.com/8676741/222039235-e1dd730e-af8f-45be-8c61-d0811ae6d386.png)

can you post more details about netbeauty version or netbeauty config or building logs?

**Comment by alexluozf** [comment]
Artifact does not exist. v6.0.15/win-x64
Thanks!

**Comment by alexluozf** [comment]
FXRVersion: v6.0.15
RID: win-x64

**Reply by liesauer** [reply]
how weird, this should exists.
![image](https://user-images.githubusercontent.com/8676741/227120106-301ce872-ea10-4f82-b0b6-315755ed89bd.png)

what is your `BeautyGitCDN` setting, and can you post the full log plz?

**Comment by alexluozf** [comment]
Hello,

It works when I enabled BeautyGitCDN setting.

Thanks a lot.

B/r

Alex Luo

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Reply by liesauer** [reply]
not at all

**Comment by kedpter** [comment]
Artifact does not exist. v6.0.16/win-x64

**Reply by kedpter** [reply]
I finally found out that v6.0.16/win-x64 (https://github.com/nulastudio/HostFXRPatcher/tree/master/artifacts/v6.0.16/win-x64.Release) exists.
 So i edit *.csproj and uncomment this line " <BeautyGitCDN>https://gitee.com/liesauer/HostFXRPatcher</BeautyGitCDN> ". 

I suggest the README should uncomment "BeautyGitCDN", cuz this problem is annoying.

**Reply by liesauer** [reply]
gitee is a mirror option for who has troble in connecting github(for example, Mainland China), it's not a good idea to make gitee as default.

**Comment by janjaapknobbout** [comment]
Artifacts do not exist. 

FXRVersion v6.0.21/linux-x64
RID linux-x64

FXRVersion v6.0.21/win-x64
RID win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by SamMercer172** [comment]
Artifacts do not exist.

FXRVersion v6.0.22/win-x64
RID win-x64

**Reply by liesauer** [reply]
uploaded

**Reply by SamMercer172** [reply]
Thanks much appreciated

**Comment by Soar360** [comment]
Artifact does not exist. v6.0.24/win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by Soar360** [comment]
Artifact does not exist. v8.0.1/win-arm64

**Reply by liesauer** [reply]
win-arm64 does not support yet

**Reply by liesauer** [reply]
uploaded

**Comment by Sergey004** [comment]
Artifact does not exist. 

FXRVersion: v8.0.2
RID: win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by Greendq** [comment]
Artifact does not exist.

FXRVersion v7.0.16/win-x64
RID win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by Apostolique** [comment]
FXRVersion: v8.0.2
RID: linux-x64

**Reply by liesauer** [reply]
uploaded

**Reply by Apostolique** [reply]
Not sure why but the artifacts for this version are missing again.

edit: Deleted `%TEMP%/NetCoreBeauty/artifacts` and it fixed the issue.

**Comment by Apostolique** [comment]
FXRVersion: v8.0.2
RID: osx-x64

**Reply by liesauer** [reply]
uploaded

**Reply by Apostolique** [reply]
Not sure why but the artifacts for this version are missing again.

**Reply by liesauer** [reply]
try this config
```xml
<BeautyGitCDN>https://gitee.com/liesauer/HostFXRPatcher</BeautyGitCDN>
```

**Reply by Apostolique** [reply]
Unfortunately didn't work either.

I also tried clearing the local package from `.nuget/packages` but I still see the same error.

> EXEC : error : Artifact does not exist. v8.0.2/osx-x64

**Reply by liesauer** [reply]
set LogLevel to `Info` and see what it says, and your NetBeauty version, please

**Reply by liesauer** [reply]
also clearing the NetBeauty data may help by deleting `%TEMP%/NetCoreBeauty`

**Reply by Apostolique** [reply]
I'm using netcorebeauty version 1.2.9.5.

I deleted `%TEMP%/NetCoreBeauty/artifacts` and it fixed the issue. (Checking the osx folder before, it was empty.)

Thanks for your help!

**Reply by liesauer** [reply]
that's good

**Comment by nygula** [comment]
Artifact does not exist. 
FXRVersion: v8.0.3
RID: win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by maxsenft** [comment]
FXRVersion: v6.0.29
RID: win-x64

FXRVersion: v8.0.4
RID: win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by neelabo** [comment]
FXRVersion:  v8.0.5
RID: win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by elmkni** [comment]
FXRVersion: v9.0.0-preview.5.24306.7/win-x64
RID: win-x64

Great program! (why didn't Microsoft come up with something like this ... ??? )

**Reply by liesauer** [reply]
uploaded

**Reply by elmkni** [reply]
Works like a charme. -- Thank You very much!

**Comment by JerryJian** [comment]
FXRVersion: v8.0.7
RID: win-x64

**Reply by liesauer** [reply]
uploaded

**Reply by JerryJian** [reply]
Thank you very much.

**Comment by GlodenBoy** [comment]
Artifact does not exist. v6.0.32/win-x86
FXRVersion:  v6.0.32 
FXR版本：v6.0.32
RID: win-x86

**Reply by liesauer** [reply]
uploaded

**Comment by Greendq** [comment]
After today`s update:
fxr v8.0.8/linux-arm64 detected 
Artifact does not exist. v8.0.8/linux-arm64

The same for other platforms.

**Reply by joesdu** [reply]
> After today`s update: fxr v8.0.8/linux-arm64 detected Artifact does not exist. v8.0.8/linux-arm64
> 
> The same for other platforms.

+1

**Reply by liesauer** [reply]
uploaded

**Comment by dmk-dlx** [comment]
FXRVersion:  v8.0.10
RID: win-x64
Artifact does not exist. v8.0.10/win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by janjaapknobbout** [comment]
Artifact does not exist. v6.0.35/win-x64

**Reply by liesauer** [reply]
uploaded

**Comment by pathartl** [comment]
v8.0.11
linux, osx, win
x64 and arm64 for all three

Caught by my CI/CD:
https://github.com/LANCommander/LANCommander/actions/runs/11812897327/job/32908899659

**Reply by liesauer** [reply]
uploaded

**Comment by neelabo** [comment]
FXRVersion: v8.0.11
RID: win-x64

The patch is not applied to hostfxr.dll But nbeauty is terminating normally.
```
patching hostfxr...
using compatible rid win-x64 for win-x64
Artifact does not exist. v8.0.11/win-x64
patch failed
:
nbeauty done. Enjoy it!
```

**Reply by liesauer** [reply]
try this solution and rebuild to see if it helps, https://github.com/nulastudio/NetBeauty2/discussions/36#discussioncomment-8591993

**Reply by neelabo** [reply]
I removed `%TEMP%/NetCoreBeauty` and the patch was applied successfully!
Thank you very much.

**Reply by liesauer** [reply]
that's good

**Comment by pathartl** [comment]
v9.0.2
linux, osx, win
x64 and arm64 for all three

https://github.com/LANCommander/LANCommander/actions/runs/13274632304/job/37061680030

**Reply by liesauer** [reply]
uploaded. btw, it's highly recommand to use the fixed verison of .NET, and upgrade it on demand.

**Comment by danielMd2805** [comment]
Hi, 
I miss the following runtime:

Artifact does not exist. v8.0.15/win-x64

Can you add it, please? :)

**Reply by liesauer** [reply]
uploaded

**Comment by Nedlow** [comment]
Hey, I miss this runtime,:

FXRVersion: v9.0.4
RID: win-x86

**Reply by liesauer** [reply]
uploaded

**Comment by JerryJian** [comment]
Hi,

FXRVersion: v8.0.16
RID: win-x64

Thanks!

**Reply by liesauer** [reply]
hey, my working pc was broken a few days ago and i don't think it can be fixed in a short time, a solution for you now is to downgrade your project's .net sdk, create `global.json` in the project root dir, and add the following content
```json
{
  "sdk": {
    "version": "8.0.408"
  }
}
```

the relationship between sdk version and fxr version you can find in here: https://dotnet.microsoft.com/en-us/download/dotnet/8.0

**Comment by Greendq** [comment]
After today`s VS update:

Artifact does not exist. v8.0.20/win-x64

Looks like the same issue with Linux/Mac platforms as well.

**Comment by Greendq** [comment]
Hi,

FXRVersion: v8.0.20
RID: win-x64

Thanks!

(I guess this is for automatic parser)

**Reply by liesauer** [reply]
uploaded.

ps: do not update vs on the first 3 days when you receive update, 99.999% missing artifacts error just gone. :smile:

**Reply by Greendq** [reply]
Thank you!

**Comment by realyw** [comment]
2>"C:\Users\liu\.nuget\packages\nulastudio.netbeauty\2.1.5\targets\../tools/win-x86/nbeauty2.exe" --gitcdn https://gitee.com/liesauer/HostFXRPatcher --gittree master --loglevel Info   --usepatch --hiddens "hostfxr;hostpolicy;*.deps.json;*.pdb"  --nbloaderverpolicy "auto"   "E:\All Project\Eding.ICU工具箱\Eding.ICU.Toolkit\bin\Release\net8.0-windows\win-x64\/." ./Libs 
2>running nbeauty...
2>checking runtime.*.json version...
2>runtime.compatibility.json no need to update
2>runtime.supported.json no need to update
2>fixing E:/All Project/Eding.ICU宸ュ叿绠?Eding.ICU.Toolkit/bin/Release/net8.0-windows/win-x64/Eding.ICU.Toolkit.deps.json
2>SCD Mode: Yes
2>SCD Version: v8.0.20, win-x64
2>Use Patch: Yes
2>Need Loader Version: No
2>ASP.NET Core: No
2>Use WPF: Yes
2>VerifyWpfDllSet: No
2>Enable Debugging: No
2>Shared Runtime Mode: No
2>E:/All Project/Eding.ICU宸ュ叿绠?Eding.ICU.Toolkit/bin/Release/net8.0-windows/win-x64/Eding.ICU.Toolkit.deps.json fixed
2>patching hostfxr...
2>using compatible rid win-x64 for win-x64
2>backuping fxr to E:\All Project\Eding.ICU宸ュ叿绠盶Eding.ICU.Toolkit\bin\Release\net8.0-windows\win-x64/hostfxr.dll.bak
2>Artifact does not exist. v8.0.20/win-x64
2>patch failed
2>fixing E:\All Project\Eding.ICU宸ュ叿绠盶Eding.ICU.Toolkit\bin\Release\net8.0-windows\win-x64\Eding.ICU.Toolkit.runtimeconfig.json
2>E:\All Project\Eding.ICU宸ュ叿绠盶Eding.ICU.Toolkit\bin\Release\net8.0-windows\win-x64\Eding.ICU.Toolkit.runtimeconfig.json fixed
2>releasing libloader.dll
2>nbeauty done. Enjoy it!

Artifact不存在依然美化完成，程序启动后有的功能有问题，还有中文路径会乱码。

**Reply by liesauer** [reply]
你的vs版本是啥

**Reply by liesauer** [reply]
另外8.0.20是有的，你使用了自定义的`BeautyGitCDN`吗？

**Reply by liesauer** [reply]
试下这个方法，把缓存清一遍再编译，https://github.com/nulastudio/NetBeauty2/discussions/36#discussioncomment-8591993

**Reply by realyw** [reply]
我的VS是最新版 17.14.16
我启用了BeautyGitCDN，我清空了%TEMP%/NetCoreBeauty文件夹，问题已解决，下载补丁已成功。
之前C:\Users\liu\AppData\Local\Temp\NetCoreBeauty\artifacts\v8.0.20\win-x64.Release文件夹是空的导致的错误。

**Reply by liesauer** [reply]
解决了就好

**Reply by realyw** [reply]
经常有些新安装的.dll库没有被移动到Libs文件夹中，然后过些天再次发布时发现又移动到Libs文件夹中了。今天又有一个新安装的SixLabors.ImageSharp.Drawing.dll发布时没有移动到Libs文件夹中，是NetBeauty2的原因吗？如何解决呢？

**Reply by realyw** [reply]
我知道咋回事了，删除obj文件夹就好了

**Reply by liesauer** [reply]
缓存机制的问题吧，每次安装、删除依赖库，都可以先清理一遍解决方案，会减少很多莫名其妙的问题

**Comment by Starciad** [comment]
fetch online runtime compatibility version failed
Artifact does not exist. v2.1.30/win-x64

**Reply by liesauer** [reply]
have you tried [`BeautyGitCDN`](https://github.com/nulastudio/NetBeauty2?tab=readme-ov-file#how-to-use)?

**Reply by Starciad** [reply]
> have you tried [`BeautyGitCDN`](https://github.com/nulastudio/NetBeauty2?tab=readme-ov-file#how-to-use)?

Ah, yes, thank you! I realized I wasn’t using it, and after adding it, everything worked as expected. 😁

**Comment by Nedlow** [comment]
Missing:

FXRVersion: v9.0.11
RID: win-x86

**Reply by liesauer** [reply]
uploaded.

**Comment by realyw** [comment]
FXRVersion: v8.0.23
RID: win-x64

**Reply by liesauer** [reply]
uploaded.

**Comment by nRoger-Env** [comment]
I think .net 10.0.3 is missing

**Reply by liesauer** [reply]
> I think .net 10.0.3 is missing

i am currently on the lunar new year holiday and may need a few more days to process this.
you can temporarily use the previous version of the sdk as a workaround.

**Reply by liesauer** [reply]
uploaded
